### PR TITLE
more async, pushing things to URIGraphSearcher and reusing derived data

### PR DESCRIPTION
**Pull request by ymatsuda**

### CODE REVIEW

**Comment by eishay** [comment]
cool!

File: shoebox/app/com/keepit/search/MainSearcherFactory.scala [hunk 1]
```diff
@@ -43,34 +48,35 @@ class MainSearcherFactory @Inject() (
...
-    val uriGraphSearcher = {
-      try {
-        uriGraph.getURIGraphSearcher(Some(userId))
-      } catch {
-        case e: URIGraphUnsupportedVersionException =>
-          // self healing, just in case
-          log.warn("fixing graph data", e)
-          uriGraph.update(userId)
+    val uriGraphSearcherFuture = consolidate(userId){ userId =>
+      future {
```
Comment (gmethvin) [inline shoebox/app/com/keepit/search/MainSearcherFactory.scala:58]:
``` scala
future {
  uriGraph.getURIGraphSearcher(Some(userId))
} recover {
  case e: URIGraphUnsupportedVersionException => ...
}
```

File: shoebox/test/com/keepit/search/graph/GraphTestHelper.scala [hunk 2]
```diff
@@ -102,6 +104,10 @@ trait GraphTestHelper extends DbRepos {
   }
 
   def mkURIGraph(graphDir: RAMDirectory = new RAMDirectory, collectionDir: RAMDirectory = new RAMDirectory): URIGraphImpl = {
-    new URIGraphImpl(mkURIGraphIndexer(graphDir), mkCollectionIndexer(collectionDir))
+    new URIGraphImpl(mkURIGraphIndexer(graphDir), mkCollectionIndexer(collectionDir), inject[ShoeboxServiceClient], inject[MonitoredAwait])
+  }
+
+  def setConnections(connections: Map[Id[User], Set[Id[User]]]) {
+    inject[ShoeboxServiceClient].asInstanceOf[FakeShoeboxServiceClientImpl].setConnections(connections)
```
Comment (gmethvin) [inline shoebox/test/com/keepit/search/graph/GraphTestHelper.scala:111]: It should be possible to do `inject[FakeShoeboxServiceClientImpl].setConnections(connections)`.

File: shoebox/app/com/keepit/common/service/RequestConsolidator.scala [hunk 3]
```diff
@@ -48,4 +48,6 @@ class RequestConsolidator[K, T](ttl: Duration) extends Logging {
         future
     }
   }
+
+  def clear() { futureRefMap.clear }
```
Comment (gmethvin) [inline shoebox/app/com/keepit/common/service/RequestConsolidator.scala]: Usually the style is to use parens (`clear()`) when a method has side effects.

**Comment by gmethvin** [comment]
looks pretty good

File: shoebox/app/com/keepit/search/MainSearcherFactory.scala [hunk 4]
```diff
@@ -82,6 +86,8 @@ class MainSearcherFactory @Inject() (
     )
   }
 
+  def clear() = consolidate.clear()
```
Comment (gmethvin) [inline shoebox/app/com/keepit/search/MainSearcherFactory.scala]: style wise, I think this should be `def clear() { consolidate.clear() }`

**Pull request merged**